163405 Enable RDPs for given IP address for both AWS/Azure

rizwancel opened the pull request.

rizwancel committed:
163405 Enable RDPs for given IP address for both AWS/Azure

rizwancel committed:
163405 Check for external IP address paramter

rizwancel committed:
163405 Add ExternalIPAddress parameter

rizwancel committed:
163405 Fix PR CI pipeline build issue

rizwancel committed:
163405 Add proper array handling of External IP Address

rizwancel committed:
163405 Properly format IP for SG update, remove whitespaces

rizwancel committed:
163405 Add valid IPs in external IP address parameter

robe070 commented on scripts/dot-CommonTools.ps1:
```diff
@@ -148,7 +148,7 @@ param (
 {
     # Wait until PSSession is available
-    $Retry = 10     # retry every 10 seconds + about 20 seconds for the timeout inherent in failing to create a new session. Typically takes 1 attempt in this loop.
+    $Retry = 30     # retry every 30 seconds + about 20 seconds for the timeout inherent in failing to create a new session. Typically takes 1 attempt in this loop.
```
No. Increase the number of loops not the time for each loop.

robe070 commented:
I've worked out the problem. The pipeline does not have access to the VM. That is, the **current** external IP must ALWAYS be added, as well as any external IPs provided in the new parameter.

rizwancel committed:
163405 Always add external IP in SG group

Enable RDP for external ip as well.
Add check for ExternalIPAddress parameter in Azure
Revert timeout value to 10 seconds and increase retry count to 10.

robe070 commented:
@rizwancel There are further changes to be made for the `VM Tests` tasks. This also creates a VM and is based on `.\scripts\Bake-IDE-msi.ps1`. The script is `.\tests\ImageVM.Tests.ps1`. Its a Pester script. You set environment variables to pass parameters in to it. E.g. $env:TestImageName.

But, it was not written knowing that pipeline variables can be accessed directly without needing an explicit environment variable. So the external ip address list can simply be accessed in the Pester Script using `$env:ExternalIPAddresses`.
A Pester Script is essentially a PowerShell script with a specialised interpreter for Pester commands. E.g. Describe and BeforeAll. But once inside the code blocks, its straight PowerShell syntax.

For AWS you can see there is a call to `Create-Ec2SecurityGroup`. So you can just pass the `$env:ExternalIPAddresses` on the end of that line.

Similarly for Azure. Do a similar change to that which you made in `.\scripts\Bake-IDE-msi.ps1`

There is also a question about when the VM gets deleted. Please work that out.

rizwancel committed:
163405 Modify VM test task and update external IP address for RDP
